### PR TITLE
Change RXMD's build system to CMake

### PR DESCRIPTION
**Pull request by TaufeqRazakh**
##### TODO:
- [x]  Update Docs
- [x] Add CMakeLists.txt in `src` dir
- [x] Modify CMakeLists.txt in `rxmdtorch`

### CODE REVIEW

**Comment by TaufeqRazakh** [comment]
Currently testing with the following build recipe from RXMD project directory (root)

```
module use /soft/modulefiles/
module load spack
module load cmake
module load PrgEnv-gnu
module load frameworks/2023.05.15.001
 
rm -rfv build
mkdir build
cd build

cmake -DCMAKE_CXX_COMPILER=icpx \
-DTorch_DIR=/soft/datascience/aurora_models_frameworks-2023.2/lib/python3.9/site-packages/torch/share/cmake/Torch/ \
-DIPEX_DIR=/soft/datascience/aurora_models_frameworks-2023.2/lib/python3.9/site-packages/intel_extension_for_pytorch/share/cmake/IPEX/ \
-DCMAKE_Fortran_COMPILER=mpif90 \
-DRXMD_ENABLE_TORCH=ON \
-DRXMD_ENABLE_IPEX=ON \
..
```

**Comment by ye-luo** [comment]
Use
```
-DTorch_ROOT=/soft/datascience/aurora_models_frameworks-2023.2/lib/python3.9/site-packages/torch
-DIPEX_ROOT=/soft/datascience/aurora_models_frameworks-2023.2/lib/python3.9/site-packages/intel_extension_for_pytorch
```
instead

File: README.md [hunk 1]
```diff
@@ -100,8 +100,16 @@ Makefile.inc  doc/          init/         rxmd*         src/          util/
...
-
-## 3. How to run
+## 3. CMake options
++ General build options
+    + `CMAKE_Fortran_COMPILER` Specify the executable for compiling Fortran files
+    + `CMAKE_CXX_COMPILER` Specify the executable for compiling C++ files 
++ RXMD build options
+    + `RXMD_ENABLE_MPI` Set this variable to on `ON` (default) to compile for distributed parallelization with MPI.  Otherwise set to `OFF`.
+    + `RXMD_ENABLE_TORCH` Set this variable to `ON` to use PyTorch libraries for machine learning tasks. Set to `OFF` by default
+    + `RXMD_ENABLE_IPEX` Set this variable to `ON` to optimized features for INtel hardware. Set to `OFF` by default.
```
Comment (ye-luo) [inline README.md]: `INtel`

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  endif()
+elseif(RXMD_ENABLE_TORCH)
+  find_package(Torch REQUIRED)
+  if(NOT TORCH_FOUND)
+    message(
+      FATAL_ERROR
+        "Torch support not found. Provide path to TorchConfig.cmake"
+    )
+  endif()  
+endif()
```
Comment (ye-luo) [inline CMakeLists.txt]: Hard to understand the above complicated if-elseif. Is ipex a drop replace ment of torch? Or it needs to be linked together with torch?

File: rxmdtorch/CMakeLists.txt [hunk 3]
```diff
@@ -1,30 +1,23 @@
...
-#####################################
-find_package(Torch REQUIRED)
-target_link_libraries(${PROJECT_NAME} "${TORCH_LIBRARIES}")
-
-if(RXMD_ENABLE_IPEX)
-  find_package(IPEX REQUIRED)
-  target_link_libraries(${PROJECT_NAME} "${TORCH_IPEX_LIBRARIES}")
+if(IPEX_FOUND)
+  target_link_libraries(torch_rxmd PUBLIC "${TORCH_IPEX_LIBRARIES}")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${TORCH_CXX_FLAGS}")
```
Comment (ye-luo) [inline rxmdtorch/CMakeLists.txt]: Is TORCH_CXX_FLAGS  defined by ipex?

File: rxmdtorch/CMakeLists.txt [hunk 4]
```diff
@@ -1,30 +1,23 @@
...
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${TORCH_IPEX_CXX_FLAGS}")
+elseif(TORCH_FOUND)
+  target_link_libraries(torch_rxmd "${TORCH_LIBRARIES}")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${TORCH_CXX_FLAGS}")
```
Comment (ye-luo) [inline rxmdtorch/CMakeLists.txt]: It is bad to directly modify CMAKE_CXX_FLAGS. Doesn't Torch and IPEX provide CMake targets?

**Review by ye-luo — APPROVED** [review]

**Pull request merged**